### DISCUSSION TITLE
Save references to objects created in instrumentation.ts

### DISCUSSION THREAD

**Discussion by zhna123** [vercel/next.js#61524] [Help]
### Summary

Hi,
I am creating a web app to schedule on/off times of some smart devices in my own home. I used a json file to store all the devices and their schedules(on/off times) information. When the app starts/re-starts, I want it to read the json file first to get all the previously scheduled data, then reschedule these jobs. I need a way to run the code to do this when app starts, but I'm having trouble implementing it.

I tried to use the `instrumentation.ts` file. It runs at app start-up, reads the json file, and schedules the job correctly. However, after my app's home page loads, the references for these scheduled jobs are lost, which means I couldn't cancel or reschedule these jobs.

Is there a way to save the references to the jobs that are scheduled in the `instrumentation.ts` file? If this is not the way to do it, can you suggest other possible ways to achieve this, preferably without using an external server?

Thanks a lot!

My web app used 
1. Next.js 14.0.4 app router
2. server actions to schedule jobs based on user input
3. [node-schedule](https://github.com/node-schedule/node-schedule) library for scheduling jobs

### Additional information

1. Next.js 14.0.4 app router
2. server actions to schedule jobs based on user input
3. [node-schedule](https://github.com/node-schedule/node-schedule) library for scheduling jobs

**Comment by Aaqu** [comment]
the problem is with instrumentation - starts after querying any endpoint, instead of when server starts
